bears/vcs/git/GitCommitBear.py

```diff
@@ -134,6 +145,30 @@ def run(self, allow_empty_commit_message: bool = False, **kwargs):
             body,
             **self.get_issue_checks_metadata().filter_parameters(kwargs))
 
+    def check_auth_email(self, auth_email,
+                         auth_email_check: bool=True,
+                         email_dns_check: bool=False):
+        """
+        Checks validity of author's email
+
+        :param auth_email_check:      If True, it will check validity of
+                                      author's email.
+        :param email_dns_check:       If True, it will confirm the domain
+                                      in author's email is valid.
+        """
+        if not auth_email_check:
+            return
+
+        if not email_dns_check:
+            email_check = is_email(auth_email)
+        else:
+            email_check = not is_email(auth_email, diagnose=True,
+                                       check_dns=True).ERROR_CODES
+
+        if not email_check:
+            yield Result(self,
```

Comment:
Please yield different results for the two possible errors. It's important for the user to know if the email is syntactically incorrect or if we can't resolve the DNS (possible network issue)